tapir/wirgarten/static/wirgarten/js/product_cfg/period_product_cfg.js

```diff
@@ -0,0 +1,322 @@
+    Array.from(listItemsProds).forEach(item =>  {
+        item.style['display'] = 'none';
+        item.classList.remove("active");
+    })
+    if (pe_pt_map[periodId] != undefined) {
+        pe_pt_map[periodId].forEach(productTypeId => {
+            const sActiveElement = "btn_prtype_" + productTypeId;
+            // sort to front to prevent artifacts
+            let elem = document.getElementById(sActiveElement);
+            let parent = elem.parentElement;
+            parent.removeChild(elem);
+            parent.prepend(elem);
+            elem.style.display = "grid";
+        });
+    }
+}
+
+const setupProductList = (productTypeId, pt_p_map_json) => {
+    const pt_p_map = JSON.parse(pt_p_map_json);
+
+    const listItems = document.getElementById("list_product").children;
+    Array.from(listItems).forEach(item =>  {
+        item.style['display'] = 'none';
+        item.classList.remove("active");
+    })
+    if (pt_p_map[productTypeId] != undefined)
+        pt_p_map[productTypeId].forEach(productId => {
+            // console.log(" -ptId: ", productTypeId, " -pId: ", productId)
+            const sActiveElement = "btn_prod_" + productId;
+            let elem = document.getElementById(sActiveElement);
+            let parent = elem.parentElement;
+            parent.removeChild(elem);
+            parent.prepend(elem);
+            elem.style.display = "grid";
+        });
+}
+
+const manageProductDependentButtons = (params) => {
+    if (params.prodId) {
+        const buttons = Array.from(document.getElementsByClassName("need-product"));
+        buttons.forEach(btn => {
+            btn.removeAttribute("disabled");
+        })
+    } else {
+        const buttons = Array.from(document.getElementsByClassName("need-product"));
+        buttons.forEach(btn => {
+            btn.setAttribute("disabled", "true");
+        })
+    }
+}
+
+const manageGrowingPeriodDependentButtons = (params) => {
+    const buttons = Array.from(document.getElementsByClassName("need-growing-period"));
+    if (params.periodId) {
+        buttons.forEach(btn => {
+            btn.removeAttribute("disabled");
+        })
+    } else {
+        buttons.forEach(btn => {
+            btn.setAttribute("disabled", "true");
+        })
+    }
+}
+
+const manageProductTypeDependentButtons = (params) => {
+    const buttons = Array.from(document.getElementsByClassName("need-product-type"));
+    if (params.prodTypeId) {
+        buttons.forEach(btn => {
+            btn.removeAttribute("disabled");
+        })
+    } else {
+        buttons.forEach(btn => {
+            btn.setAttribute("disabled", "true");
+        })
+    }
+}
+
+const manageButtons = () => {
+    const params = getUrlParams();
+    manageGrowingPeriodDependentButtons(params);
+    manageProductTypeDependentButtons(params);
+    manageProductDependentButtons(params);
+}
+
+// dynamic Forms
+const getProductTypeEditForm = () => {
```

Comment:
lots of duplicated code from here on, the largest part of the get*Form() function can be easily extracted into a new function